home/views.py

```diff
@@ -13,16 +14,29 @@ def landingpage(request):
     return render(request, 'home/landingpage.html')
 
 
-def displayQuestion(request, **kwargs):
-    question = get_object_or_404(Question, id=kwargs['pk'])
-    sortAnsBy = request.GET["sortanswersby"] if 'sortanswersby' in request.GET else ''
+def display_question_page(request, pk):
+    form = CommentForm(request.POST)
+    question = get_object_or_404(Question, id=pk)
+    sortAnsBy = request.GET.get("sortanswersby", "")
     context = {
         "question": question,
         "answers": question.get_answers_feed(sortAnsBy),
         "answersCount": question.answer_set.count(),
         "tags": question.tags.values(),
-        "title": question.get_question_title()
+        "title": question.get_question_title(),
+        "form": form
     }
+    if request.method == 'POST':
+        if form.is_valid():
+            if (request.profile is None):
+                raise Http404()
```

Comment:
Maybe switch it to 401 response - unauthorized. (Barak recommended it to me also in a similar case)
you can see how I did it in my PR in views.py

Reply:
you right. 404 is really doesn't the right status code for this case